solver/src/main/java/org/chocosolver/util/objects/IVal.java

```diff
@@ -25,4 +27,100 @@ public interface IVal {
     void update(int nb_probes);
 
     void transfer();
+
+    /**
+     *
+     * <p>
+     * Project: choco-solver.
+     * @author Charles Prud'homme
+     * @since 30/10/2018.
+     */
+    class MapVal implements IVal {
+
+        private final TIntDoubleHashMap Av;
+        private final TIntDoubleHashMap mAv;
+        private final int os;  // offset
+
+        public MapVal(int os) {
+            this.os = os;
+            this.Av = new TIntDoubleHashMap(32, 0.5f, 0, 0);
+            this.mAv = new TIntDoubleHashMap(32, 0.5f, 0, 0);
+        }
+
+        @Override
+        public double activity(int value) {
+            return Av.get(value - os);
+        }
+
+        @Override
+        public void setactivity(int value, double activity) {
+            Av.put(value - os, activity);
+        }
+
+        @Override
+        public void update(int nb_probes) {
+            double activity, oldmA, U;
+            for (int k : Av.keys()) {
+                activity = Av.get(k);
+                oldmA = mAv.get(k);
+                U = activity - oldmA;
+                mAv.adjustValue(k, U / nb_probes);
+            }
+        }
+
+        @Override
+        public void transfer() {
+            Av.clear();
+            Av.putAll(mAv);
+        }
+    }
+
+    /**
+     * <p>
+     * Project: choco-solver.
+     *
+     * @author Charles Prud'homme
+     * @since 30/10/2018.
+     */
+    class ArrayVal implements IVal {
```

Comment:
Here too, do we need various implementations (open question, I have no idea) ?

Reply:
Thanks! 
It is not really the purpose of the PR but as I see it is only used by ABS, I would like to suggest a few changes : 
- Renamming it to something more accurate, maybe IActivityTracker, IActivityStore...  I find IVal a bit too generic, values are manipulated all over the solver.
- Moving IVal to a specific abs package where is the activitybasedsearch in order to keep only generic and reusable objects in org.chocosolver.util.objects;
- Extracting implementations into separate files